cpp/src/arrow/filesystem/azurefs.h

```diff
@@ -112,6 +112,9 @@ struct ARROW_EXPORT AzureOptions {
   /// This will be ignored if non-empty metadata is passed to OpenOutputStream.
   std::shared_ptr<const KeyValueMetadata> default_metadata;
 
+  /// Whether OutputStream writes will be issued in the background, without blocking.
+  bool background_writes = true;
```

Comment:
How about referring `ARROW_ENABLE_THREADING` here?

```suggestion
#ifdef ARROW_ENABLE_THREADING
  bool background_writes = true;
#else
  bool background_writes = false;
#endif
```

Reply:
Shouldn't this be set to `false` in the beginning before potential bugs are ironed out?

Reply:
Fine for me, lets start with disabling it per default at first.